File: spring_webapp1/src/main/java/com/example/spring_webapp1/application/printing/PrintReceiptUseCase.java
package com.example.spring_webapp1.application.printing;

import com.example.spring_webapp1.domain.printing.IReceiptPrinter;
import com.example.spring_webapp1.domain.printing.ReceiptDocument;
import org.springframework.stereotype.Component;

import java.awt.print.PrinterException;

/**
 * レシート印刷ユースケース。
 * テキストを解析して {@link ReceiptDocument} を作り、{@link IReceiptPrinter} に委譲する。
 */
@Component
public class PrintReceiptUseCase {

    private final IReceiptPrinter printer;
    private final ReceiptTextParser parser;

    public PrintReceiptUseCase(IReceiptPrinter printer, ReceiptTextParser parser) {
        this.printer = printer;
        this.parser = parser;
    }

    public void execute(String text, String printerKeyword) throws PrinterException {
        execute(text, printerKeyword, null);
    }

    public void execute(String text, String printerKeyword, String logoPath) throws PrinterException {
        ReceiptDocument document = parser.parse(text, logoPath);
        printer.printAndCut(document, printerKeyword);
    }
}

File: MauiApp1/Application/Printing/PrintReceiptUseCase.cs
using MauiApp1.Domain.Printing;

namespace MauiApp1.Application.Printing;

/// <summary>
/// レシート印刷ユースケース。
/// テキストを解析して <see cref="ReceiptDocument"/> を作り、<see cref="IReceiptPrinter"/> に委譲する。
/// </summary>
public sealed class PrintReceiptUseCase
{
    private readonly IReceiptPrinter _printer;

    public PrintReceiptUseCase(IReceiptPrinter printer)
        => _printer = printer;

    public Task ExecuteAsync(
        string text,
        string? logoPath = null,
        CancellationToken cancellationToken = default)
    {
        var document = ReceiptTextParser.Parse(text, logoPath);
        return _printer.PrintAndCutAsync(document, null, cancellationToken);
    }
}
